### ISSUE TITLE
py.path.local.visit breaks when using "from __future__ import unicode_literals"

### ISSUE THREAD

**Issue by pytestbot** [pytest-dev/py#81]
- Bitbucket: https://bitbucket.org/pytest-dev/py/issue/81
- Originally reported by: [montefra](http://bitbucket.org/montefra)
- Originally created at: 2016-07-07T10:42:38.484

For python 2<->3 compatibility I write all my code with:

```
#!python

from __future__ import (absolute_import, division, print_function,
                                      unicode_literals)
```

at the beginning at each file. 
In python 2 this breaks the visit method in py.path.common  because [these three lines](https://bitbucket.org/pytest-dev/py/src/4617fe465cfc3289cefb8e7b8ce28ed5168f139f/py/_path/common.py?at=default&fileviewer=file-view-default#common.py-348) only check if `fil` and `rec` are string.

The solution is either to use `isinstance(fil, six.string_types)` (if you add `six` as dependency) or to do something like:

```
#!python
class Visitor:
    def __init__(self, fil, rec, ignore, bf, sort):
        if sys.version_info >= (3, ):
            string_types = str
        else:
             string_types = basestring
        if isinstance(fil, string_types):
            fil = FNMatcher(fil)
        if isinstance(rec, string_types):
            self.rec = FNMatcher(rec)
```

**Comment by montefra** [comment]
if you want a PR I can do it on Monday

**Comment by RonnyPfannschmidt** [comment]
@montefra pylib has a own version of string_types (its support libs predate six)

i'd love to see a pr

**Comment by montefra** [comment]
@RonnyPfannschmidt : I'll do on Monday. Have a nice weekend

**Comment by RonnyPfannschmidt** [comment]
closed by #86
